components/admin/layers/form/LayersForm.js

```diff
@@ -116,39 +124,34 @@ class LayersForm extends React.Component {
 
       const form = Object.assign({}, this.state.form, { interactionConfig });
 
+      // Verify that layers are valid, otherwise render error
+      const { adminLayerPreview } = this.props;
+      const cartoLayer = adminLayerPreview.layerGroups[0].layers
+      .filter(layer => layer.provider === 'cartodb');
+
       if (valid) {
-        // if we are in the last step we will submit the form
-        if (this.state.step === this.state.stepLength && !this.state.submitting) {
-          const { id, dataset } = this.state;
-
-          // Start the submitting
-          this.setState({ submitting: true });
-
-          this.service.saveData({
-            dataset,
-            id: id || '',
-            type: (id) ? 'PATCH' : 'POST',
-            body: form
-          })
-            .then((data) => {
-              toastr.success('Success', `The layer "${data.id}" - "${data.name}" has been uploaded correctly`);
-              if (this.props.onSubmit) this.props.onSubmit();
-            })
-            .catch((errors) => {
+        // Start the submitting
+        this.setState({ submitting: true });
+        this.props.dispatch(setLayerInteractionError(false));
+
+        if (cartoLayer.length) {
```

Comment:
Should we think in the future? I mean, right now we only have cartodb interactions, but in the future we will add all kinds of layers. Should we have a utils file to verify every kind of layer? Don't do this right now, but next time it would be great